vital/tests/test_mesh.cxx

```diff
@@ -0,0 +1,236 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/// \file
+/// \brief test core mesh functionality
+
+#include <tests/test_gtest.h>
+#include <tests/test_scene.h>
+#include <vital/exceptions.h>
+
+#include <vital/io/mesh_io.h>
+
+#include <gtest/gtest.h>
+
+kwiver::vital::path_t g_data_dir;
+
+// ----------------------------------------------------------------------------
+int main(int argc, char** argv)
+{
+  ::testing::InitGoogleTest( &argc, argv );
+  return RUN_ALL_TESTS();
+}
+
+// ----------------------------------------------------------------------------
+void compare_meshes(const kwiver::vital::mesh_sptr& first,
+                    const kwiver::vital::mesh_sptr& second)
+{
+  kwiver::vital::mesh_vertex_array<3>& first_vertices = dynamic_cast
+    <kwiver::vital::mesh_vertex_array<3>&>( first->vertices() );
+  kwiver::vital::mesh_vertex_array<3>& second_vertices = dynamic_cast
+    <kwiver::vital::mesh_vertex_array<3>&>( second->vertices() );
+
+  EXPECT_EQ( first_vertices.size(), second_vertices.size() );
+  for(unsigned int i=0; i<first_vertices.size(); i++)
+  {
+    EXPECT_EQ( first_vertices[i].size(), second_vertices[i].size() );
+    for(unsigned int j=0; j<first_vertices[i].size(); j++)
+    {
+      EXPECT_EQ( first_vertices[i][j], second_vertices[i][j] );
+    }
+  }
+
+  kwiver::vital::mesh_face_array first_faces = *std::make_shared
+    <kwiver::vital::mesh_face_array>(first->faces());
+  kwiver::vital::mesh_face_array second_faces = *std::make_shared
+    <kwiver::vital::mesh_face_array>(second->faces());
+
+  EXPECT_EQ( first_faces.size(), second_faces.size() );
+  for(unsigned int i=0; i<first_faces.size(); i++)
+  {
+    EXPECT_EQ( first_faces[i].size(), second_faces[i].size() );
+    for(unsigned int j=0; j<first_faces[i].size(); j++)
+    {
+      EXPECT_EQ( first_faces[i][j], second_faces[i][j] );
+    }
+  }
+}
+
+// ----------------------------------------------------------------------------
+TEST(mesh, group_names)
```

Comment:
Since this is really targeted to test the mesh_face_array class, should the test name be "TEST( mesh_face_array, group_names )" instead, or should it stay with just "mesh", as that's the name of the file?